config/configgrpc/configgrpc.go

```diff
@@ -400,7 +400,11 @@ func contextWithClient(ctx context.Context, includeMetadata bool) context.Contex
 	}
 	if includeMetadata {
 		if md, ok := metadata.FromIncomingContext(ctx); ok {
-			cl.Metadata = client.NewMetadata(md.Copy())
+			copiedMD := md.Copy()
+			if len(md["Host"]) == 0 && len(md[":authority"]) > 0 {
+				copiedMD["Host"] = md[":authority"]
+			}
```

Comment:
should we remove the ":authority" in this case? (Not sure, but want to make sure somebody thinks about this).

Reply:
I think no, because it's real header

Reply:
Agree, at least as part of this PR here. We might want to start a discussion for something like "semantic conventions" for the headers.

Reply:
I was agreeing in keeping `:authority` here, as it's the key used by gRPC, and would expect to see it there. The second part of my comment was about having a single value where downstream components would expect the "host" value to exist, but this is being addressed in this PR already. Sorry about the noise.

Reply:
> "semantic conventions" for the headers.

This also may include us having the constant values for things like "Host", that's why I got confused. So you are ok with having that constant?

Reply:
Yes, at least for the moment. If we keep finding ourselves doing it often, we might want to split that out.